File: app/src/main/java/black/bracken/shotsorter/util/SimpleTextWatcher.java
package black.bracken.shotsorter.util;

import android.text.Editable;
import android.text.TextWatcher;

/**
 * @author BlackBracken
 */
public interface SimpleTextWatcher extends TextWatcher {

    default void beforeTextChanged(CharSequence charSequence, int start, int count, int after) {}

    default void afterTextChanged(Editable editable) {}

}

File: app/src/main/java/black/bracken/shotsorter/util/SimpleTextWatcher.kt
package black.bracken.shotsorter.util

import android.text.Editable
import android.text.TextWatcher

/**
 * @author BlackBracken
 */
class SimpleTextWatcher(
        private val action: (CharSequence, Int, Int, Int) -> Unit
) : TextWatcher {

    override fun onTextChanged(charSequence: CharSequence, start: Int, count: Int, after: Int) = action(charSequence, start, count, after)

    override fun beforeTextChanged(charSequence: CharSequence, start: Int, count: Int, after: Int) {}

    override fun afterTextChanged(editable: Editable) {}

}
